tests/unit/test_results.py

```diff
@@ -41,6 +42,48 @@ def _get_vcr(self, **kwargs):
             "Set-Cookie",
             "User-Agent",
         ]
+        myvcr.filter_query_parameters = [
+            ("client_id", "foo"),
+        ]
+
+        def redact_key_values(keys_to_redact):
+            def before_record_response(response):
+                # Only do this if the response has not been recorded
+                string_body = response["body"]["string"].decode("utf8")
+                if "REDACTED" not in string_body:
+                    # Only do this is if the body contains one or more
+                    # of the keys to redact.
+                    if any(key in string_body for key in keys_to_redact):
+                        try:
+                            # Marshall into json object, if it is a JSON object.
+                            payload = json.loads(string_body)
+                            if isinstance(payload, list):
+                                payload = payload[0]
```

Comment:
Is there a particular situation where we expect the payload to be a list? If so, why would we take only the first element of the list?

Reply:
Some responses from EDL are JSON wrapped in a list which is not parsing via json.loads. This is a hack which I'd like some python expertise on. How do I handle this elegantly?

Reply:
Ah, I see that you're getting back a list from the /tokens endpoint, which makes sense. I see the problem you're facing, so let me see if I can find a convenient means for inserting my suggested code adjustments.